File: HideBloat/src/main/java/com/aliucord/plugins/hidebloat/views/ChannelsInviteButtonPatch.java
package com.aliucord.plugins.hidebloat.views;

import android.view.View;
import android.view.ViewGroup;

import com.aliucord.plugins.hidebloat.Const;
import com.aliucord.plugins.hidebloat.views.base.BasePatcher;
import com.discord.widgets.channels.list.WidgetChannelsListAdapter;
import com.discord.widgets.channels.list.items.ChannelListItem;

import top.canyie.pine.Pine;

public class ChannelsInviteButtonPatch extends BasePatcher {

    private static final String WIDGET_CHANNELS_LIST_ADAPTER_ITEM_INVITE = "com.discord.widgets.channels.list.WidgetChannelsListAdapter$ItemInvite";
    private static final String ON_CONFIGURE = "onConfigure";

    public ChannelsInviteButtonPatch() {
        super(Const.CHANNELS_INVITE_BUTTON_KEY, WIDGET_CHANNELS_LIST_ADAPTER_ITEM_INVITE, ON_CONFIGURE, new Class[] { int.class, ChannelListItem.class });
    }

    @Override
    public void patchBody(Pine.CallFrame callFrame) {
        View itemView = ((WidgetChannelsListAdapter.ItemInvite) callFrame.thisObject).itemView;
        assert itemView != null;
        itemView.setVisibility(View.GONE);

        //Adjust LayoutParams, especially height, otherwise list
        //will include padding of the button with it's original size
        itemView.setLayoutParams(new ViewGroup.LayoutParams(0, 0));
        callFrame.setResult(callFrame.getResult());
    }
}

File: plugins/LayoutController/src/main/java/com/xinto/aliuplugins/layoutcontroller/patchers/ChannelsInviteButtonPatch.kt
package com.xinto.aliuplugins.layoutcontroller.patchers

import com.xinto.aliuplugins.layoutcontroller.patchers.base.BasePatcher
import com.xinto.aliuplugins.layoutcontroller.util.Description
import com.xinto.aliuplugins.layoutcontroller.util.Key
import com.xinto.aliuplugins.layoutcontroller.util.hideCompletely
import com.discord.widgets.channels.list.WidgetChannelsListAdapter
import com.discord.widgets.channels.list.items.ChannelListItem
import de.robv.android.xposed.XC_MethodHook

class ChannelsInviteButtonPatch : BasePatcher(
    key = Key.INVITE_BUTTON_CHANNELS_KEY,
    description = Description.INVITE_BUTTON_CHANNELS_DESCRIPTION,
    requiresRestart = false,
    classMember = WidgetChannelsListAdapter.ItemInvite::class.java.getDeclaredMethod(
        "onConfigure",
        Int::class.javaPrimitiveType,
        ChannelListItem::class.java
    )
) {
    override fun patchBody(callFrame: XC_MethodHook.MethodHookParam) {
        val itemView = (callFrame.thisObject as WidgetChannelsListAdapter.ItemInvite).itemView
        itemView.hideCompletely()
        callFrame.result = callFrame.result
    }
}